### ISSUE TITLE
Unable to connect to position (AFV API Issue?)

### ISSUE THREAD

**Issue by dechilders** [pierr3/VectorAudio#5]
Hi Pierre,

I've got VectorAudio installed and configured, and I'm connected to the network (through VRC). When I try to connect VectorAudio, the callsign I'm connected as does not auto-add as it should (at least from what I understand). When I try to manually add it, it doesn't do anything. The afv.log file shows a couple of errors related to the API.

[Log File](https://github.com/pierr3/VectorAudio/files/8145536/afv.txt)

**Comment by pierr3** [comment]
This seems to be an issue with your vatsim credentials. Can you triple check the settings tab in VectorAudio with your CID and password ? 

After you've connected with VRC, it might take up to 45 seconds to detect your connection.

**Comment by pierr3** [comment]
I suspect this is something I didn't foresee, the search API in AFV might not actually be public, my CID has extra permissions which means it works. Let me investigate this.

**Comment by dechilders** [comment]
I double checked the credentials and everything seems fine, it shows my callsign in the top left before I even connect, so I'm assuming that means the credentials are for sure right. I'm able to connect and add other positions in my ARTCC, but when I try to add mine nothing happens.

Will wait to see what you find on the search API, thanks!

**Comment by pierr3** [comment]
Ah so you can add other positions aside from the one you are connected as? Could you share which callsign you're trying to add? Can you check you're running the latest version/commit?

**Comment by jonaseberle** [comment]
I think @dechilders you expected that your logged in station is auto-added.

That is not the case and I haven't expected VECTOR to do that yet.

After connecting I am adding my station (or ones I want to monitor) and they get added then.
![image](https://user-images.githubusercontent.com/1678001/155844833-ee3e6040-187c-4480-94f6-a5aeb8cce95c.png)

**Comment by dechilders** [comment]
I can add other callsigns, but not my own. The only callsign I have tested it with for so far is SDF_GND. I tried entering stuff like IND_GND and IND_CTR and both populate correctly like I would expect them to, but nothing for SDF_GND.

I'm running the latest version (0.2.0), I was actually watching the GitHub action waiting on it to finish so I could grab the latest artifact it produced. I tested an earlier build (before the 0.2.0 produced last night), by pulling the source and compiling it myself, but it had the same results.

**Comment by pierr3** [comment]
Haha I really need to put some caching on those github actions, sorry about how crazy long it takes. 

I've just done a search in the AFV database, and it seems like SDF_GND does not exist, this might be an issue for your FE to add. Currently VectorAudio assumes all frequencies are in the database, if you try to add a frequency which is not defined in AFV, it won't add it. I need to add an airport database, so that in these situations VectorAudio can create an ad-hoc frequency.

To recap, IND_GND is in the AFV database, SDF_GND is not, which is why you cannot add it. Can you ask your FE to define it?

**Comment by pierr3** [comment]
> I think @dechilders you expected that your logged in station is auto-added.
> 
> That is not the case and I haven't expected VECTOR to do that yet.
> 
> After connecting I am adding my station (or ones I want to monitor) and they get added then. ![image](https://user-images.githubusercontent.com/1678001/155844833-ee3e6040-187c-4480-94f6-a5aeb8cce95c.png)

The auto-add feature has been added in v0.2.0 :)

**Comment by dechilders** [comment]
> To recap, IND_GND is in the AFV database, SDF_GND is not, which is why you cannot add it. Can you ask your FE to define it?

Ahh that makes sense. I will get with the staff at my ARTCC and see if they can add it. We don't currently have a FE, but I'm sure someone can get it done. Thanks for looking into it!

**Comment by pierr3** [comment]
If someone from your ARTCC shoots me a discord DM or email, I can also add it if you have no one currently managing this :)

**Comment by dechilders** [comment]
Apparently anything TWR down is not added to the AFV database by design and they won't be adding it, thanks for the help though Pierre!

**Comment by pierr3** [comment]
Thanks for letting me know! This is actually quite important, because it's a different behaviour to the classic AFV client that I need to implement. I'll add the same feature means I need to add an airport AFV and automatically create the frequency

**Comment by jonaseberle** [comment]
> The auto-add feature has been added in v0.2.0 :)

Is that meant to work like that?

* I connect with my radar client and a primary frequency set
* When I start VECTOR and connect, my station's definition is already in the list

In that case that has not worked for me either with VECTIR v2.0 with `EDDS_N_APP`. 
When manually adding it, that station and all below (`DEL`, `GND`, `TWR` and secondary `APP` frequencies) show up.

**Comment by pierr3** [comment]
I've checked and indeed it should be working, but as suspected I am using a restrict AFV API call. I've asked the AFV dev to open up the API and it should fix it
